tests/lean/run/coinductive_predicates.lean

```diff
@@ -0,0 +1,147 @@
+-- Coinductive predicate definition
+def infseq {α} (R : α → α → Prop) : α → Prop :=
+  λ x : α => ∃ y, R x y ∧ infseq R y
+  greatest_fixpoint
+
+-- Application of the rewrite rule
+def infseq_fixpoint {α} (R : α → α → Prop) (x : α) :
+  infseq R x = ∃ y, R x y ∧ infseq R y := by
+    rw [infseq]
+
+-- Usage of the associated coinduction principle
+theorem infseq.coind {α} (h : α → Prop) (R : α → α → Prop)
+  (prem : ∀ (x : α), h x → ∃ y, R x y ∧ h y) : ∀ x, h x → infseq R x := by
+  apply infseq.fixpoint_induct
+  exact prem
```

Comment:
If this test is supposed to check the type of `infseq.fixpoint_induct`, then don’t write a theorem that uses it, but simply use
```
#guard_msgs in #check infseq.fixpoint_induct
```
to document the expected type.